tests/phpunit/CRM/WeAct/Page/StripeTest.php

```diff
@@ -0,0 +1,258 @@
+<?php
+
+use CRM_WeAct_ExtensionUtil as E;
+
+/**
+ * @group headless
+ */
+class CRM_WeAct_Page_StripeTest extends CRM_WeAct_BaseTest {
+
+  public function setUp() : void {
+    parent::setUp();
+  }
+
+  public function testNewRecurringPayment() {
+    $subscription = "sub_AccmDyDhCXVvJtXf";
+    $action = CRM_WeAct_Action_ProcaTest::recurringStripeAction('monthly', NULL, $subscription);
+    $processor = new CRM_WeAct_ActionProcessor();
+    $processor->processDonation($action, $this->campaignId, $this->contactId);
+
+    $invoice = "in_sZJpHDMwEnahHziF";
+    $charge = "ch_rgWvkHQrADdcgPdY";
+    $payment_intent = "pi_uFoDjxBJLjaZpqBn";
+
+    $created = '2022-01-28 01:34:00';
+    $created_dt = new DateTime($created);
+
+    $stripe_event = $this->recurringPayment(
+      $subscription,
+      $invoice,
+      $charge,
+      $payment_intent,
+      $created_dt->getTimestamp()
+    );
+
+    $page = new CRM_WeAct_Page_Stripe();
+    $page->processNotification(json_decode($stripe_event));
+
+    $contribution = civicrm_api3('Contribution', 'getsingle', ['trxn_id' => $invoice]);
+    $this->assertEquals($contribution['receive_date'], $created);
+    $this->assertEquals($contribution['contribution_status_id'], 1); # Completed
+  }
+
+  public function testUnknownEvent() {
+    $page = new CRM_WeAct_Page_Stripe();
+    $event = <<<JSON
+      { "type": "invoice.idontexist" }
+JSON;
+    $page->processNotification(json_decode($event));
+    $this->assertTrue(true);
+  }
+
+  // TODO: Unknown subscription - But what should it do? Create the subscription right?
+  // public function testUnknownRecurringDonation() {}
+
+  // TODO: Updated subscription
+  // public function testUpdatedRecurringSubscription() {}
+
+  protected function recurringPayment($subscription,$invoice,$charge,$payment_intent,$created) {
+    return <<<JSON
```

Comment:
One practice I follow is to remove from the mock event all the stuff that's never going to be used by the code. It makes it easier what the mock event is about, easier to maintain, and "documents" what's used by the logic. What do you think?

Reply:
Done!